File: Java/Class-2.4/src/org/launchcode/main/FinalFields.java
package org.launchcode.main;

public class FinalFields {

	public final int intValue = 42;

	// Not allowed
	// public final double doubleValue;
	public final FortyTwo objectValue = new FortyTwo();

	public static void main(String[] args) {

		FinalFields demo = new FinalFields();

		// This would result in a compiler error
		// demo.intValue = 6;

		// This is allowed since we haven't initialized doubleValue yet
		// demo.doubleValue = 42.0;

		// However, this would result in a compiler error
		// demo.doubleValue = 6.0;

		// This would result in a compiler error, since we're trying to
		// give objectValue a different object value
		// demo.objectValue = new FortyTwo();

		// However, this is allowed since we're changing a field
		// inside the final object, and not changing which object
		// objectValue refers to
		// demo.objectValue.intValue = 6;
	}

	// Note: We can have two classes in one file!
	public class FortyTwo {
		public int intValue = 42;

	}
}

File: CSharp/LC101-Unit2/Class-2.5/FinalFields.cs
using System;
namespace Class_2._5
{
    public class FinalFields
    {
        // Final fields!
        // Primitives cannot be changed
        // Objects (classes) can have values within the object changed (by reference), but not the object itself
        public readonly int intValue = 42;
        public readonly double doubleValue;
        public readonly FortyTwo objectValue = new FortyTwo();

        public FinalFields()
        {
        }
    }
}
